### ISSUE TITLE
Two regularization terms in insetGAN

### ISSUE THREAD

**Issue by fan84sunny** [afruehstueck/insetGAN#7]
Hi. I have read your paper in CVPR, and I have some question about this project.

In paper, the two regularization terms are 
$L_{reg}=\left\| w^{*} - w_{avg}\  \right\|+\sum \left\| \delta_i \right\|$

I only can find the first term. Since the second term is not used in the project, I don't know how it works about δ.
https://github.com/afruehstueck/insetGAN/blob/47f80f4b0564029122bd2dde75fd315e58ecf953/run_insetgan.py#L128
Can you describe the function works and what are X_mean, X_comp, X_stdev?
Thanks.

**Comment by afruehstueck** [comment]
You are correct that the second regularization term is not used in this particular implementation, for simplicity. 
The first part of the term $\lambda_{r1} || w^* - w_{avg} ||$ is computed in  [  `loss = loss_fn(latent_in, latent_avg_canvas.clone())` ](https://github.com/afruehstueck/insetGAN/blob/47f80f4b0564029122bd2dde75fd315e58ecf953/run_insetgan.py#L349)
As for the second term, we are able to generate slightly better and more expressive results when decomposing the term 
[`latent_in = latent_w_canvas.unsqueeze(0).repeat(G_canvas.num_ws, 1).unsqueeze(0)`  ](https://github.com/afruehstueck/insetGAN/blob/47f80f4b0564029122bd2dde75fd315e58ecf953/run_insetgan.py#L244)
into two independently optimized terms, `latent_w_canvas` and `latent_w_canvas_offset`, where `latent_w_canvas` would have dimensionality `(n, 512)`, and `latent_w_canvas_offset` would have dimensionality `(n, 512, num_ws)`.
`latent_in` would therefore be
`latent_in = latent_w_canvas.unsqueeze(0).repeat(G_canvas.num_ws, 1).unsqueeze(0) + latent_w_canvas_offset` and `latent_w_canvas_offset` would be optimized jointly with `latent_w_canvas`, and regularized to be small with a L2 loss. This allows a (small) divergence from $w$ space without exhausting the full capacity of $w^+$ space.

The term you are referencing is a remnant from legacy code and, I believe, unused in the code.
